scroll_overlay/android/app/src/main/java/io/flutter/scroll_overlay/ScrollOverlayActivity.java

```diff
@@ -3,151 +3,190 @@
 import android.app.Activity;
 import android.graphics.drawable.ColorDrawable;
 import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
 import android.util.TypedValue;
 import android.view.Gravity;
-import android.view.MotionEvent;
+import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.BaseAdapter;
-import android.widget.ListView;
 import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.DividerItemDecoration;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import io.flutter.embedding.android.FlutterView;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterEngineCache;
+import io.flutter.embedding.engine.dart.DartExecutor;
 import io.flutter.plugin.common.EventChannel;
-import io.flutter.view.FlutterMain;
-import io.flutter.view.FlutterRunArguments;
-import io.flutter.view.FlutterView;
 
 public class ScrollOverlayActivity extends Activity {
+    private static final String VELOCITY_CHANNEL = "scroll_overlay.flutter.io/velocity";
+    private static final int[] OVERLAY_COLORS = new int[]{0x40ff0000, 0x4000ff00, 0x400000ff};
+    private static final String engineId = "engine_id";
 
-    private static final String TAG = "ScrollOverlayActivity";
+    FlutterEngine flutterEngine;
+    EventChannel.EventSink velocitySink;
 
-    private static final String VELOCITY_CHANNEL = "scroll_overlay.flutter.io/velocity";
+    class VelocityTracker extends RecyclerView.OnScrollListener {
+        int currentDy = 0;
+        boolean locked = false;
 
-    private static final int[] OVERLAY_COLORS = new int[]{0x40ff0000, 0x4000ff00, 0x400000ff};
+        final int ticksPerSecond = 25;
```

Comment:
how is this value determined?

Reply:
I picked it arbitrarily and it looked good. Do you have any concerns about this?

Reply:
maybe try explaining what this value does, and why 25 is a good value for it

Reply:
done - renamed the variable and added comments both in native and dart sides